packages/ec-core/src/operator/selector/weighted.rs

```diff
@@ -48,3 +48,32 @@ where
         selector.select(population, rng)
     }
 }
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use test_strategy::proptest;
+
+    use super::Weighted;
+    use crate::operator::selector::{best::Best, worst::Worst, Selector};
+
+    #[proptest]
+    fn test_random(#[any] values: [i32; 10]) {
+        let pop: Vec<i32> = values.into();
+        let mut rng = rand::thread_rng();
+        // We'll make a selector that has a 50/50 chance of choosing the highest
+        // or lowest value.
+        let weighted = Weighted::new(Best, 1).with_selector(Worst, 1);
+        let selection = weighted.select(&pop, &mut rng).unwrap();
+        let smallest = values.iter().min().unwrap();
+        let largest = values.iter().max().unwrap();
```

Comment:
```suggestion
        let (smallest,largest) = values.iter().minmax().into_option().unwrap();
```
may be nicer, minmax is from itertools.